src/main/java/com/aws/greengrass/localdebugconsole/SimpleHttpServer.java

```diff
@@ -154,13 +218,54 @@ public void startup() throws InterruptedException {
         reportState(State.RUNNING);
     }
 
+    static String fingerprintCert(X509Certificate cert)
+            throws NoSuchAlgorithmException, CertificateEncodingException {
+        StringBuilder sb = new StringBuilder();
+        for (byte b : MessageDigest.getInstance("SHA-256").digest(cert.getEncoded())) {
+            sb.append(String.format("%02X ", b));
+        }
+        return sb.toString().trim();
+    }
+
+    private static X509Certificate selfSign(KeyPair keyPair, String subjectDN) throws OperatorCreationException,
+            CertificateException,
+            IOException
+    {
+        java.security.Provider bcProvider = new BouncyCastleProvider();
+        Security.addProvider(bcProvider);
+
+        long now = System.currentTimeMillis() - TimeUnit.SECONDS.toMillis(60);
+        Date startDate = new Date(now);
+
+        X500Name dnName = new X500Name("cn=" + subjectDN);
+        BigInteger certSerialNumber = new BigInteger(Long.toString(now)); // Using the current timestamp as the certificate serial number
+
+        Calendar calendar = Calendar.getInstance();
+        calendar.setTime(startDate);
+        calendar.add(Calendar.YEAR, 1); // 1 Year validity
+        Date endDate = calendar.getTime();
+
+        String signatureAlgorithm = "SHA256WithRSA";
+        ContentSigner contentSigner = new JcaContentSignerBuilder(signatureAlgorithm).build(keyPair.getPrivate());
+        JcaX509v3CertificateBuilder
+                certBuilder = new JcaX509v3CertificateBuilder(dnName, certSerialNumber, startDate,
+                endDate, dnName, keyPair.getPublic());
+        BasicConstraints basicConstraints = new BasicConstraints(false);
+        // Required basic constraints OID. Must be present to be recognized as a proper certificate by browsers
+        certBuilder.addExtension(new ASN1ObjectIdentifier("2.5.29.19"), true, basicConstraints);
+
+        return new JcaX509CertificateConverter().setProvider(bcProvider).getCertificate(certBuilder.build(contentSigner));
+    }
+
     @Override
     public void shutdown() throws InterruptedException {
         logger.atInfo().log("Shutting down httpd");
         secondaryGroup.shutdownGracefully();
         primaryGroup.shutdownGracefully();
         try {
-            dashboardServer.stop();
+            if (dashboardServer != null) {
+                dashboardServer.stop();
+            }
         } catch (Exception e) {
             logger.atError().setCause(e).log("Error shutting down local dashboard server");
             serviceErrored(e);
```

Comment:
do you need to log here? I noticed there isn't a log in some of the other `serviceErrored(e)` calls